watch-library/shared/watch/watch_private_display.c

```diff
@@ -87,10 +88,60 @@ void watch_display_character(uint8_t character, uint8_t position) {
         segmap = segmap >> 8;
         segdata = segdata >> 1;
     }
+}
+
+void watch_display_character(uint8_t character, uint8_t position) {
+    watch_display_segdata(watch_convert_char_to_segdata(character, position), position);
 
     if (character == 'T' && position == 1) watch_set_pixel(1, 12); // add descender
-    else if (position == 0 && (character == 'B' || character == 'D')) watch_set_pixel(0, 15); // add funky ninth segment
-    else if (position == 1 && (character == 'B' || character == 'D' || character == '@')) watch_set_pixel(0, 12); // add funky ninth segment
+    else if (position == 1 && (character == 'B' || character == 'D' || character == 'b' || character == 'd' || character == '@')) watch_set_pixel(0, 12); // add funky ninth segment
+    else if (position == 0 && (character == 'B' || character == 'D')) {
+        // add funky ninth segment
+        watch_set_pixel(0, 15);
+    } else {
+        watch_clear_pixel(0, 15);
+    }
+}
+
+void watch_display_string_morph(char *old_string, char *new_string) {
+    uint8_t old_segdata[Num_Chars];
+    uint8_t new_segdata[Num_Chars];
+    size_t i;
+    size_t seg_len;
+
+    for (i = 0; old_string[i] && new_string[i] && i < Num_Chars; ++i) {
+        old_segdata[i] = watch_convert_char_to_segdata(old_string[i], i);
+        new_segdata[i] = watch_convert_char_to_segdata(new_string[i], i);
+    }
+
+    seg_len = i;
+
+    // Make old_segdata approach new_segdata, one bit at a time.
+    for (i = 0; i < seg_len; ++i) {
+        if (old_segdata[i] == new_segdata[i]) {
+            continue;
+        }
+
+        for (int bit_pos = 0; bit_pos < 8; ++bit_pos) {
+            int bit = 1 << bit_pos;
+
+            if ((old_segdata[i] & bit) != (new_segdata[i] & bit)) {
+                old_segdata[i] ^= bit;
+
+                uint64_t segmap = Segment_Map[i] >> (8 * bit_pos);
```

Comment:
Should probably pull this back to the start of the loop, as in the normal display. Originally, I was choosing which bits to flip on a different basis, but left->right actually looked the best (IMO).